whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy

```diff
@@ -1002,6 +1013,18 @@ class PostgreSQLComponent {
             
             if (collection == "hold") {
                 checkLinkedShelfMarkOwnership(doc, connection)
+
+                String heldBy = doc.getHeldBy()
+                if (heldBy == null) {
+                    log.warn("Was asked to save a holding record lacking a correct library/heldBy (so, did nothing).")
+                    return null
+                }
+                String libraryRecordId = getRecordId(heldBy, connection)
+                if (libraryRecordId == null) {
+                    log.warn("Was asked to save a holding record for a library that could not be located: " + heldBy + " (so, did nothing).")
+                    return null
+                }
```

Comment:
Should it be possible to delete documents with bad heldBy links?
Now it is not possible without first changing heldBy to a valid value.

Reply:
You are right! I'll fix it!